### PR TITLE
Add CodeQL Security Scanning

### PR DESCRIPTION
**Pull request by caniszczyk**
Signed-off-by: Chris Aniszczyk <caniszczyk@gmail.com>

### CODE REVIEW

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [ 'go', 'javascript' ]
```
Comment (sargun) [inline .github/workflows/codeql-analysis.yml:31]: Can you remove javascript?

File: .github/workflows/codeql-analysis.yml [hunk 2]
```diff
@@ -0,0 +1,67 @@
+# For most projects, this workflow file will not need changing; you simply need
```
Comment (sargun) [inline .github/workflows/codeql-analysis.yml:1]: I think you can remove this note.

**Review by sargun — CHANGES_REQUESTED** [review]
I'm not totally sure what CodeQL does, and where the results of these actions go. Can you add a little bit to your commit comment about that?

**Comment by sargun** [comment]
@caniszczyk Ping?

File: .github/workflows/codeql-analysis.yml (hunk 2 above)
Comment (caniszczyk) [reply .github/workflows/codeql-analysis.yml]: removed the note

File: .github/workflows/codeql-analysis.yml (hunk 1 above)
Comment (caniszczyk) [reply .github/workflows/codeql-analysis.yml]: done

**Pull request merged**